lib/puppet/provider/package/yum.rb

```diff
@@ -25,6 +32,34 @@
 defaultfor :operatingsystem => :amazon
 defaultfor :osfamily => :redhat, :operatingsystemmajrelease => (4..7).to_a
 
+  def insync?(is)
+    return false if [:purged, :absent].include?(is)
+    return false if is.include?(self.class::MULTIVERSION_SEPARATOR) && !@resource[:install_only]
+    should = @resource[:ensure]
+
+    if should.is_a?(String)
+      begin
+        should_version = RPM_VERSION_RANGE.parse(should, RPM_VERSION)
+      rescue RPM_VERSION_RANGE::ValidationFailure, RPM_VERSION::ValidationFailure
+        Puppet.debug("Cannot parse #{should} as a RPM version range")
+      end
+   
+      begin
+        is_version = RPM_VERSION.parse(is)
+      rescue RPM_VERSION::ValidationFailure
+        Puppet.debug("Cannot parse #{is} as a RPM version")
+      end
+   
+      if should_version && is_version
+        return should_version.include?(is_version)
+      else
+        is.split(self.class::MULTIVERSION_SEPARATOR).any? do |version|
+          0 == rpm_compareEVR(rpm_parse_evr(should), rpm_parse_evr(version))
+        end
```

Comment:
In order to handle multiversion packages with VersionRange, we should run the check inside `any?`, like:
``` 
      is.split(self.class::MULTIVERSION_SEPARATOR).any? do |version|
        begin
          is_version = RPM_VERSION.parse(is)
          should_version.include?(is_version)
        rescue RPM_VERSION::ValidationFailure
          Puppet.debug("Cannot parse #{is} as a RPM version")
        end
      end
```